ExampleMod/Content/Items/ExampleInstancedItem.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.ModLoader;
+using Terraria.ModLoader.IO;
+
+namespace ExampleMod.Content.Items
+{
+	public class ExampleInstancedItem : ModItem
+	{
+		public override string Texture => "ExampleMod/Content/Items/ExampleItem";
+
+		public override ModItem Clone(Item item) {
+			ExampleInstancedItem clone = (ExampleInstancedItem)base.Clone(item);
+			clone.colors = (Color[])colors.Clone();
+			return clone;
+		}
+
+		public Color[] colors;
+
+		public override void OnCreate(ItemCreationContext context) {
+			colors = new Color[5];
+			for (int i = 0; i < 5; i++) {
+				colors[i] = Main.hslToRgb(Main.rand.NextFloat(), 1f, 0.7f);
+			}
+		}
+
+		public ExampleInstancedItem() {
+			colors = new Color[5];
+			for (int i = 0; i < 5; i++) {
+				colors[i] = Main.hslToRgb(Main.rand.NextFloat(), 1f, 0.7f);
+			}
```

Comment:
this is the kind of duplication we want to avoid. We could call `OnCreate(null)` here, but ideally we wouldn't use the constructor at all. The only places where `OnCreate` isn't called is in declarative templates like recipes or loot tables. Maybe we should make a creation context for this? Unsure